src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -42,6 +43,69 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return _hasPowerlineCharacters.value_or(false);
     }
 
+    Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> Font::FontAxesTagsAndNames()
+    {
+        auto fontAxesTagsAndNames = winrt::single_threaded_map<winrt::hstring, winrt::hstring>();
+
+        winrt::com_ptr<IDWriteFont> font;
+        THROW_IF_FAILED(_family->GetFont(0, font.put()));
+        winrt::com_ptr<IDWriteFontFace> fontFace;
+        THROW_IF_FAILED(font->CreateFontFace(fontFace.put()));
+        winrt::com_ptr<IDWriteFontFace5> fontFace5{ fontFace.as<IDWriteFontFace5>() };
+
+        winrt::com_ptr<IDWriteFontResource> fontResource;
+        THROW_IF_FAILED(fontFace5->GetFontResource(fontResource.put()));
+
+        std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
+        const auto axesCount = fontFace5->GetFontAxisValueCount();
+        if (axesCount > 0)
+        {
+            uint32_t localeIndex;
+            BOOL localeExists;
+            wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+            const auto localeToTry = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH) ? localeName : L"en-US";
+
+            axesVector.resize(axesCount);
+            fontFace5->GetFontAxisValues(axesVector.data(), axesCount);
+            for (uint32_t i = 0; i < axesCount; ++i)
+            {
+                const auto tagString = _axisTagToString(axesVector[i].axisTag);
+
+                winrt::com_ptr<IDWriteLocalizedStrings> names;
+                THROW_IF_FAILED(fontResource->GetAxisNames(i, names.put()));
+
+                if (!SUCCEEDED(names->FindLocaleName(localeToTry, &localeIndex, &localeExists)) || !localeExists)
+                {
+                    // default to the first locale in the list
+                    localeIndex = 0;
+                }
+
+                uint32_t length;
+                if (SUCCEEDED(names->GetStringLength(0, &length)))
+                {
+                    wchar_t* name = new (std::nothrow) wchar_t[length + 1];
```

Comment:
There has to be a better way to initialize `name`

Reply:
There are a number of options, but I think the most important one would be to ensure that the allocated memory is being freed again. One option would be
```cpp
std::make_unique_for_overwrite<wchar_t>(length + 1);
```

Personally speaking, I believe it would also be reasonable to assume a fixed upper bound for name lengths. For instance a fixed buffer of 512 characters should cover any valid translation but only consumes 1KiB of stack space. I.e.:

```cpp
for (...)
{
    wchar_t name[512];
    if (SUCCEEDED(names->GetString(0, &name[0], ARRAYSIZE(name))))
    {
        fontAxesTagsAndNames.Insert(tagString, winrt::hstring{ &name[0] });
        continue;
    }
}
```

Both approaches are fine. This isn't a hot path after all so we don't need to be conservative with heap allocations.

Reply:
Makes sense, fixed